libs/dmlf/include/dmlf/collective_learning/client_algorithm.hpp

```diff
@@ -168,7 +174,9 @@ void ClientAlgorithm<TensorType>::SetModel(ModelPtrType model_ptr)
 template <class TensorType>
 void ClientAlgorithm<TensorType>::ClearLossFile()
 {
-  std::ofstream lossfile("losses_" + id_ + ".csv", std::ofstream::out | std::ofstream::trunc);
+  mkdir(params_.results_dir.c_str(), 0777);
```

Comment:
i would guess that this is not very cross platform friendly? please check with me that it works at least on macos and ubuntu

Reply:
It's POSIX only, so i'll work on mac and linux